drivers/ieee802154/ieee802154_mcxw.c

```diff
@@ -540,10 +549,21 @@ static int mcxw_tx(const struct device *dev, enum ieee802154_tx_mode mode, struc
 #if defined(CONFIG_NET_PKT_TXTIME)
 	case IEEE802154_TX_MODE_TXTIME:
 	case IEEE802154_TX_MODE_TXTIME_CCA:
-		mcxw_radio->tx_frame.tx_delay = net_pkt_timestamp_ns(pkt);
+		mcxw_radio->tx_frame.tx_delay = net_pkt_timestamp_ns(pkt) / NSEC_PER_USEC;
 		msg->msgData.dataReq.startTime =
 			rf_adjust_tstamp_from_app(mcxw_radio->tx_frame.tx_delay);
-		msg->msgData.dataReq.startTime /= IEEE802154_SYMBOL_TIME_US;
+
+		/* Compensates the 192µs SHR duration by adjusting the MAC payload start
+		 * time backwards and converting it to symbol periods.
+		 */
+		if ((msg->msgData.dataReq.startTime > IEEE802154_SHR_DURATION_US) &&
```

Comment:
start time expiration is done on the NBU side by the PHY.
There is no need to do any validation here